administrator-documentation/moderne-platform/how-to-guides/lst-cleanup.md

```diff
@@ -0,0 +1,76 @@
+Be very careful before running the GraphQL query mentioned in this section. Depending on the parameters provided, it can cause some LSTs to be unavailable for up to 24 hours. There is no way to cancel or stop this query once it has been started.
+{% endhint %}
+
+Moderne provides a [GraphQL reindex query that you can run to force a sync of all LSTs](https://app.moderne.io/graphql?url=https%253A%252F%252Fapi.app.moderne.io%252Fgraphql\&query=bXV0YXRpb24gcmVpbmRleCB7CiAgaW5kZXgoZm9yY2VVcGRhdGU6IHRydWUsIG1vZGlmaWVkU2luY2UgOiIyMDIzLTAxLTAxVDAwOjAwWiIpewogICAgY291bnQKICB9Cn0%3D).
+
+{% tabs %}
+{% tab title="Reindex mutation" %}
+```graphql
+mutation reindex {
+  index(forceUpdate: true, modifiedSince :"2023-01-01T00:00Z"){
+    count
+  }
+}
+```
+{% endtab %}
+
+{% tab title="cURL" %}
+{% code overflow="wrap" %}
+```bash
+curl -X POST https://api.app.moderne.io/graphql \
+    -H 'Authorization: Bearer <session token or Moderne PAT here>' \
+    -H 'Content-Type: application/json' \
+    -d '{ "query": "mutation reindex {\n  index(forceUpdate: true, modifiedSince :\"2023-01-01T00:00Z\"){\n    count\n  }\n}" }'
+```
+{% endcode %}
+{% endtab %}
+{% endtabs %}
+
+Running this `reindex` query will cause the agent to reach out to your artifact repository and ask for all available LSTs. It will then download the metadata for each of these. After all of the metadata has been downloaded, the old LSTs will effectively be removed from the platform. Moderne will then begin downloading the latest LSTs for each repository. As this runs, you will see repositories begin appearing in the platform again.
+
+While you can still run recipes on the LSTs that are coming in, all of the LSTs won't be available for a significant amount of time (up to 24 hours depending on the number of repositories you have). Because of that, you should strongly consider only running this query during off-hours or on a weekend when there won't be much traffic.
+
+### More targeted, custom code
+
+If you want to ensure Moderne removes old LSTs without incurring any downtime, you can write custom code to interact with the Moderne GraphQL APIs. This code would probably take the form of:
+
+* Get the set of all artifacts for all repositories that exist in Moderne
+* Ask your artifact repository if those still exist over there
+* If they don't, make a request to remove those LSTs from the Moderne platform
+
+You could then run this custom script once a week or so.
+
+## Cleaning up old LSTs for branches that still exist
+
+In this scenario, you won't notice any issues in Moderne. You will still be able to run recipes against the latest LSTs and commit the results from said recipes. However, your artifact repository will begin to fill up with old LSTs that are no longer needed.
+
+To fix this issue, we encourage you to write some form of automation that removes old LSTs from your artifact repository (perhaps by deleting all LSTs that haven't been updated in over a week).
```

Comment:
Would be great to include an example AQL query to find old LSTs. I don't know if anyone has one on hand or not, though. I presume we do for our own LST storage. Maybe Nate or Scott know.